crates/noirc_frontend/src/hir/def_collector/errors.rs

```diff
@@ -29,6 +30,8 @@ pub enum DefCollectorErrorKind {
     PathResolutionError(PathResolutionError),
     #[error("Non-struct type used in impl")]
     NonStructTypeInImpl { span: Span },
+    #[error("Non-struct type used in trait impl")]
+    NonStructTraitImpl { trait_ident: Ident, span: Span },
```

Comment:
Non-struct types are allowed in trait impls. The trait should just be defined in the current crate if so.

Reply:
We've added this to the roadmap, https://github.com/noir-lang/noir/issues/2568

The issue preventing us from allowing this right away, is that structs have their own module, in which the trait methods get declared, which I believe is not the case for say primitive types, or typedefs